src/App.tsx

```diff
@@ -1,61 +1,129 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoWithUser, User } from './react-app-env';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUser = (userId: number): User | null => {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+};
+
+const todosWithUser: TodoWithUser[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const getTodoId = (todos: TodoWithUser[]) => {
+  const maxTodoId = Math.max(...todos.map(todo => todo.id));
+
+  return maxTodoId + 1;
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState<TodoWithUser[]>(todosWithUser);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [hasTitleError, setTitleError] = useState(false);
+  const [todoUserId, setTodoUserId] = useState(0);
+  const [hasUserIdError, setUserIdError] = useState(false);
+
+  const resetForm = () => {
+    setTodoTitle('');
+    setTodoUserId(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimTitle = todoTitle.trim();
+
+    setTitleError(!trimTitle);
+    setUserIdError(!todoUserId);
+
+    if (!trimTitle || !todoUserId) {
+      return;
+    }
+
+    const newTodo: TodoWithUser = {
+      id: getTodoId(todos),
+      title: todoTitle,
+      completed: false,
+      userId: todoUserId,
+      user: getUser(todoUserId),
+    };
+
+    if (!hasTitleError && !hasUserIdError) {
```

Comment:
```suggestion
```
You don't need this check. Look line 45 - it's the same.

Reply:
ok